client/rest-high-level/src/test/java/org/elasticsearch/client/BulkProcessorIT.java

```diff
@@ -268,23 +283,89 @@ public void testBulkProcessorConcurrentRequestsReadOnlyIndex() throws Exception
         assertMultiGetResponse(highLevelClient().mget(multiGetRequest, RequestOptions.DEFAULT), testDocs);
     }
 
-    private static MultiGetRequest indexDocs(BulkProcessor processor, int numDocs) throws Exception {
+    @SuppressWarnings("unchecked")
+    public void testGlobalParametersAndBulkProcessor() throws Exception {
+        createIndexWithShards("test", 2);
```

Comment:
it has to be bigger then 1 to make the routing parameter testable. Truth be spoken - the bigger the better. I will make it random in around of 10 maybe as otherwise it still will hit the same shard.